google-beta/pubsub_utils.go

```diff
@@ -0,0 +1,32 @@
+package google
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func comparePubsubSubscriptionBasename(_, old, new string, _ *schema.ResourceData) bool {
```

Comment:
nit: Since this isn't really pubsub specific, I'd slightly prefer this to live in `self_link_helpers` as something like `compareNameInStateToSelfLink`

Not necessary, so don't bother if you can confirm ackDeadlineSeconds works as expected & aren't making another pass.